addons/account/models/account_move.py

```diff
@@ -3503,7 +3503,16 @@ def _get_starting_sequence(self):
         last_month = int(self.company_id.fiscalyear_last_month)
         is_staggered_year = last_month != 12 or last_day != 31
         if is_staggered_year:
-            if self.date > date(self.date.year, last_month, last_day):
+            try:
+                compare_date = date(self.date.year, last_month, last_day)
+            except ValueError as e:
+                # If date() fails and is not February 29th,
+                # then the provided date is not valid.
+                if last_day != 29 or last_month != 2:
+                    raise e
+                compare_date = date(self.date.year, last_month, last_day - 1)
+
+            if self.date > compare_date:
```

Comment:
```suggestion
            if self.date > date(self.date.year, last_month, 1) + relativedelta(day=last_day):
```
Instead of the try-except block, how about using relativedelta? It will take care of leap year issue and it's simpler :) Perhaps adding a comment too could be nice

Reply:
Ohhh, yeah, I like that suggestion! Will make the changes and send for re-review ^u^